Onboarding with Mattermost Blog Post

imisshtml opened the pull request:
#### Summary
Blog post about my onboarding with Mattermost

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-21130

amyblais commented on site/content/blog/2019-12-20-onboarding-with-mattermost.md:
```diff
@@ -0,0 +1,34 @@
+### Sponsored Simulation
+The onboarding process at Mattermost was as unique as the company itself. As the company is fully remote, they took a great deal of care in making sure that not only was I a good fit, but that I would enjoy the environment and process that they instill. The first step was through a sponsored simulation, where I was given the opportunity to go beyond the normal open-source contributions and work d ...
+
+During the process, my mentor would help guide me in architecting the overall solution and in answering any questions I had. The community for Mattermost was also available, where I could post my questions to the core team and the open-source community. While I did not participate in any Scrum-related activities with the teams, I did follow their build and pull-request flow. The work overall was  ...
```
```suggestion
During the process, my mentor helped guide me in architecting the overall solution and in answering any questions I had. The community for Mattermost was also available, where I could post my questions to the core team and the open-source community. While I did not participate in any Scrum-related activities with the teams, I did follow their build and pull request flow. The work overall was very fast-paced and responses to my questions and reviews of my PRs were surprisingly quick. Considering everyone is remote, I was expecting the pace to be slower, but was pleasantly surprised that I was helped and guided more quickly than I would have expected in a normal office environment.
```

amyblais replied:
```suggestion
During the process, my mentor would help guide me in architecting the overall solution and in answering any questions I had. The community instance for Mattermost was also available, where I could post my questions to the core team and the open-source community. While I did not participate in any Scrum-related activities with the teams, I did follow their build and pull-request flow. The work overall was very fast-paced and responses to my questions and reviews of my PRs were surprisingly quick. Considering everyone is remote I was expecting the pace to be slower but was pleasantly surprised that I was helped and guided more quickly than I would have expected in a normal office environment.
```

amyblais commented on site/content/blog/2019-12-20-onboarding-with-mattermost.md:
```diff
@@ -0,0 +1,34 @@
+During the process, my mentor would help guide me in architecting the overall solution and in answering any questions I had. The community for Mattermost was also available, where I could post my questions to the core team and the open-source community. While I did not participate in any Scrum-related activities with the teams, I did follow their build and pull-request flow. The work overall was  ...
+
+### Detailed Checklist and Follow-Up
+One of the first things to occur is my mentor and a member of Human Resources created an extensive checklist for me to follow. This took me through all the starting documentation I would need to know and introduced me to videos and our development processes. This took me through my first two-weeks and made it easy to keep on track as I learned more about my duties and the company as a whole. 
```
```suggestion
One of the first things that occurred was that my mentor and a member of Human Resources created an extensive checklist for me to follow. This took me through all the starting documentation I would need to know and introduced me to videos and our development processes. This took me through my first two-weeks and made it easy to keep on track as I learned more about my duties and the company as a whole. 
```

amyblais commented on site/content/blog/2019-12-20-onboarding-with-mattermost.md:
```diff
@@ -0,0 +1,34 @@
+There is a vast wealth of documentation available through Confluence. Everything from setting up your development environment and the Tick-Tock branching process to the setup and customization of the servers, apps and plugins. Many of these are part of the onboarding checklist, but it will take weeks to ingest all the knowledge!
+
+### Mentorship
+During my first few weeks I would get daily mentors, from different teams, who would be available for discussion and to help answer questions that I would have. This was a great way to build a bond with my peers and to learn more about the other teams. The Mattermost application extends across web, mobile, server side, Redux, and design; in meeting with the mentors from each team I was able to be ...
```
```suggestion
During my first few weeks I had daily mentors from different teams who were available for discussions and helped answer any questions that I had. This was a great way to build a bond with my peers and to learn more about the other teams. The Mattermost application extends across web, mobile, server side, Redux, and design; in meeting with the mentors from each team I was able to better understand how all the parts of the application come together. 
```

amyblais commented on site/content/blog/2019-12-20-onboarding-with-mattermost.md:
```diff
@@ -0,0 +1,34 @@
+### Mentorship
+During my first few weeks I would get daily mentors, from different teams, who would be available for discussion and to help answer questions that I would have. This was a great way to build a bond with my peers and to learn more about the other teams. The Mattermost application extends across web, mobile, server side, Redux, and design; in meeting with the mentors from each team I was able to be ...
+
+Beyond the daily mentorships, I also had my main mentor to always follow up with and to help me through my daily tasks. One of the mindsets that was repeatedly conveyed was ‘Learn, Master, Teach’, where you learn a topic, master it, and then teach it to others; allowing the cycle to continue. 
```
```suggestion
Beyond the daily mentorships, I also had my main mentor to always follow up with and help me through my daily tasks. One of the mindsets that was repeatedly conveyed was ‘Learn, Master, Teach’, where you learn a topic, master it, and then teach it to others, allowing the cycle to continue. 
```

amyblais commented on site/content/blog/2019-12-20-onboarding-with-mattermost.md:
```diff
@@ -0,0 +1,34 @@
+Beyond the daily mentorships, I also had my main mentor to always follow up with and to help me through my daily tasks. One of the mindsets that was repeatedly conveyed was ‘Learn, Master, Teach’, where you learn a topic, master it, and then teach it to others; allowing the cycle to continue. 
+
+### Remote Environment
+One of the biggest draws of joining Mattermost is that everyone is remote, and the company has really built a culture around this. Every week there are multiple meetings that help create rapport among the team and instill the passion behind innovation and customer obsession. Unlike previous remote environments, the environment that Mattermost has created makes you feel part of the team and not si ...
```
```suggestion
One of the biggest draws of joining Mattermost is that everyone is remote, and the company has really built a culture around this. Every week there are multiple meetings that help create rapport among the team and instill the passion behind innovation and customer obsession. Unlike my previous remote environments, the environment that Mattermost has created makes you feel part of the team and not siloed off. There are even video meetings for hanging around the water cooler to just chat as well as a video game hour.
```

amyblais commented on site/content/blog/2019-12-20-onboarding-with-mattermost.md:
```diff
@@ -0,0 +1,34 @@
+One of the biggest draws of joining Mattermost is that everyone is remote, and the company has really built a culture around this. Every week there are multiple meetings that help create rapport among the team and instill the passion behind innovation and customer obsession. Unlike previous remote environments, the environment that Mattermost has created makes you feel part of the team and not si ...
+
+### Overall
+These first months here have really been amazing and through meeting the talented people who work here, the amazing open-source community, and the depth of knowledge available; I know I'm in an environment that will challenge me and help me grow as a developer. Everyone I've met has a strong passion to make our product better and to enhance the experience for our customers. This is conveyed daily ...
```
```suggestion
These first months here have really been amazing. Through meeting the talented people who work here, the amazing open-source community, and the depth of knowledge available, I know that I'm in an environment that will challenge me and help me grow as a developer. Everyone I've met has a strong passion to make our product better and to enhance the experience for our customers. This is conveyed daily through the actions and attitude of every member here and I'm excited and thankful I've had this opportunity to be part of something this incredible.
```

jasonblais commented on site/content/blog/2019-12-20-onboarding-with-mattermost.md:
```diff
@@ -0,0 +1,34 @@
+### Remote Environment
+One of the biggest draws of joining Mattermost is that everyone is remote, and the company has really built a culture around this. Every week there are multiple meetings that help create rapport among the team and instill the passion behind innovation and customer obsession. Unlike my previous remote environments, the environment that Mattermost has created makes you feel part of the team and not ...
+
+### Overall
```
```suggestion
### Summary
```

0/5 to be honest, but I thought 'summary' might fit better here as a section heading. Another heading I often see at the end of a blog is 'conclusion', but I thought you're not really 'concluding' anything here, per say.

jasonblais commented on site/content/blog/2019-12-20-onboarding-with-mattermost.md:
```diff
@@ -0,0 +1,34 @@
+One of the first things that occurred was that my mentor and a member of Human Resources created an extensive checklist for me to follow. This took me through all the starting documentation I would need to know and introduced me to videos and our development processes. This took me through my first two-weeks and made it easy to keep on track as I learned more about my duties and the company as a  ...
+
+### Extensive Resources
+There is a vast wealth of documentation available through Confluence. Everything from setting up your development environment and the Tick-Tock branching process to the setup and customization of the servers, apps and plugins. Many of these are part of the onboarding checklist, but it will take weeks to ingest all the knowledge!
```
In addition to docs in Confluence, are there other docs you found helpful? Such as those in developers.mattermost.com, docs.mattermost.com or handbook.mattermost.com?

jasonblais commented on site/content/blog/2019-12-20-onboarding-with-mattermost.md:
```diff
@@ -0,0 +1,34 @@
+
+I recently joined Mattermost as a Remote Software Engineer and while I’ve worked in remote environments before, I was really surprised at how amazing this company is to work with. During my onboarding I discovered they have really developed a culture built around being Remote; one that allowed for me to feel as part of a larger team and always engaged, rather than separated from my peers and work ...
+
+### Sponsored Simulation
```
For each section, would there be a public reference / link / page with more details? I'm thinking that if someone would like to learn more about the sponsored simulation, or mentorship, then the reader can click the link to learn more.

E.g. for sponsored simulation, we could link to https://docs.mattermost.com/process/developer.html#audition. For onboarding/checklist, we could link to https://handbook.mattermost.com/contributors/onboarding/staff-on-boarding-guide or https://developers.mattermost.com/internal/onboarding/new-staff-guide/ (not sure which is used). I can search for other resources as well, but wasn't sure which ones you had primarily used.

(If a page or link doesn't exist, no need to create one - we can just use whatever we have today).

jasonblais commented on site/content/blog/2019-12-20-onboarding-with-mattermost.md:
```diff
@@ -0,0 +1,34 @@
+
+During the process, my mentor would help guide me in architecting the overall solution and in answering any questions I had. The community instance for Mattermost was also available, where I could post my questions to the core team and the open-source community. While I did not participate in any Scrum-related activities with the teams, I did follow their build and pull-request flow. The work ove ...
+
+### Detailed Checklist and Follow-Up
```
```suggestion
### Detailed Onboarding Checklist and Follow-Up
```

imisshtml commented:
Thanks @amyblais and @jasonblais for the review! I loved the feedback and made the updates, thank you for the keen eye!

hanzei commented:
I'm wondering if `@justinegeffen` should glance over this. She did review the other recent blog posts and did a great job doing so.

Thoughts?

justinegeffen commented on site/content/blog/2019-12-20-onboarding-with-mattermost.md:
```diff
@@ -0,0 +1,34 @@
+community: claude.wild
+---
+
+I recently joined Mattermost as a Remote Software Engineer and while I’ve worked in remote environments before, I was really surprised at how amazing this company is to work with. During my onboarding I discovered they have really developed a culture built around being Remote; one that allowed for me to feel as part of a larger team and always engaged, rather than separated from my peers and work ...
```
```suggestion
I recently joined Mattermost as a Remote Software Engineer and while I’ve worked in remote environments before, I was really surprised at how amazing this company is to work with. During my onboarding I discovered they have really developed a culture built around being remote; one that allowed for me to feel as part of a larger team and always engaged, rather than separated from my peers and working alone. 
```

justinegeffen commented on site/content/blog/2019-12-20-onboarding-with-mattermost.md:
```diff
@@ -0,0 +1,34 @@
+### [Sponsored Simulation](https://docs.mattermost.com/process/developer.html#audition)
+The onboarding process at Mattermost was as unique as the company itself. As the company is fully remote, they took a great deal of care in making sure that not only was I a good fit, but that I would enjoy the environment and process that they instill. The first step was through a sponsored simulation, where I was given the opportunity to go beyond the normal open-source contributions and work d ...
+
+During the process, my mentor would help guide me in architecting the overall solution and in answering any questions I had. The community instance for Mattermost was also available, where I could post my questions to the core team and the open-source community. While I did not participate in any Scrum-related activities with the teams, I did follow their build and pull-request flow. The work ove ...
```
```suggestion
During the process, my mentor would help guide me in architecting the overall solution and in answering any questions I had. The community instance for Mattermost was also available, where I could post my questions to the core team and the open source community. While I did not participate in any Scrum-related activities with the teams, I did follow their build and pull-request flow. The work overall was very fast-paced and responses to my questions and reviews of my PRs were surprisingly quick. Considering everyone is remote I was expecting the pace to be slower but was pleasantly surprised that I was helped and guided more quickly than I would have expected in a normal office environment.
```

justinegeffen commented on site/content/blog/2019-12-20-onboarding-with-mattermost.md:
```diff
@@ -0,0 +1,34 @@
+During the process, my mentor would help guide me in architecting the overall solution and in answering any questions I had. The community instance for Mattermost was also available, where I could post my questions to the core team and the open-source community. While I did not participate in any Scrum-related activities with the teams, I did follow their build and pull-request flow. The work ove ...
+
+### [Detailed Onboarding Checklist and Follow-Up](https://handbook.mattermost.com/contributors/onboarding/staff-on-boarding-guide)
+One of the first things that occurred was that my mentor and a member of Human Resources created an extensive checklist for me to follow. This took me through all the starting documentation I would need to know and introduced me to videos and our development processes. This took me through my first two-weeks and made it easy to keep on track as I learned more about my duties and the company as a  ...
```
```suggestion
One of the first things that occurred was that my mentor and a member of Human Resources created an extensive checklist for me to follow. This took me through all the starting documentation I would need to know and introduced me to videos and our development processes. This took me through my first two weeks and made it easy to keep on track as I learned more about my duties and the company as a whole. 
```

justinegeffen commented on site/content/blog/2019-12-20-onboarding-with-mattermost.md:
```diff
@@ -0,0 +1,34 @@
+One of the first things that occurred was that my mentor and a member of Human Resources created an extensive checklist for me to follow. This took me through all the starting documentation I would need to know and introduced me to videos and our development processes. This took me through my first two-weeks and made it easy to keep on track as I learned more about my duties and the company as a  ...
+
+### [Extensive Resources](https://docs.mattermost.com/)
+There is a vast wealth of documentation available through the [Mattermost Documentation](https://docs.mattermost.com/), [Developer Documentation](https://developers.mattermost.com/), and their [Handbook](https://handbook.mattermost.com/). Everything from setting up your development environment and the Tick-Tock branching process to the setup and customization of the servers, apps and plugins. Man ...
```
```suggestion
There is a wealth of documentation available through the [Mattermost Documentation](https://docs.mattermost.com/), [Developer Documentation](https://developers.mattermost.com/), and their [Handbook](https://handbook.mattermost.com/). Everything from setting up your development environment and the Tick-Tock branching process to the setup and customization of the servers, apps, and plugins. Many of these are part of the onboarding checklist, but it will take weeks to ingest all the knowledge!
```

justinegeffen commented on site/content/blog/2019-12-20-onboarding-with-mattermost.md:
```diff
@@ -0,0 +1,34 @@
+One of the biggest draws of joining Mattermost is that everyone is remote, and the company has really built a culture around this. Every week there are multiple meetings that help create rapport among the team and instill the passion behind innovation and customer obsession. Unlike my previous remote environments, the environment that Mattermost has created makes you feel part of the team and not ...
+
+### Summary
+These first months here have really been amazing. Through meeting the talented people who work here, the amazing open-source community, and the depth of knowledge available, I know that I'm in an environment that will challenge me and help me grow as a developer. Everyone I've met has a strong passion to make our product better and to enhance the experience for our customers. This is conveyed dai ...
```
```suggestion
These first months here have really been amazing. Through meeting the talented people who work here, the amazing open source community, and the depth of knowledge available I know that I'm in an environment that will challenge me and help me grow as a developer. Everyone I've met has a strong passion to make our product better and to enhance the experience for our customers. This is conveyed daily through the actions and attitude of every member here and I'm excited and thankful I've had this opportunity to be part of something this incredible.
```

justinegeffen commented:
> I'm wondering if `@justinegeffen` should glance over this. She did review the other recent blog posts and did a great job doing so.
> 
> Thoughts?

Thanks for the tag! I fixed a few minor typos/punctuation errors but otherwise it's a great blog post. :)

imisshtml commented:
> > I'm wondering if `@justinegeffen` should glance over this. She did review the other recent blog posts and did a great job doing so.
> > Thoughts?
> 
> Thanks for the tag! I fixed a few minor typos/punctuation errors but otherwise it's a great blog post. :)

Thanks @justinegeffen, I've updated the noted items!

amyblais commented on site/content/blog/2019-12-20-onboarding-with-mattermost.md:
```diff
@@ -0,0 +1,34 @@
+community: claude.wild
+---
+
+I recently joined Mattermost as a Remote Software Engineer and while I’ve worked in remote environments before, I was really surprised at how amazing this company is to work with. During my onboarding I discovered they have really developed a culture built around being remote; one that allowed for me to feel as part of a larger team and always engaged, rather than separated from my peers and work ...
```
```suggestion
I recently joined Mattermost as a Remote Software Engineer and while I’ve worked in remote environments before, I was really surprised at how amazing this company is to work with. During my onboarding I discovered that they have really developed a culture built around being remote; one that allowed for me to feel as part of a larger team and always engaged, rather than separated from my peers and working alone. 
```

amyblais commented on site/content/blog/2019-12-20-onboarding-with-mattermost.md:
```diff
@@ -0,0 +1,34 @@
+
+Beyond the daily mentorships, I also had my main mentor to always follow up with and help me through my daily tasks. One of the mindsets that was repeatedly conveyed was ‘Learn, Master, Teach’, where you learn a topic, master it, and then teach it to others, allowing the cycle to continue.  
+
+### [Remote Environment](https://docs.mattermost.com/process/handbook.html?highlight=remote#company)
```
```suggestion
### [Remote Environment](https://docs.mattermost.com/process/handbook.html)
```

jasonblais commented:
I think this one looks ready to merge then :) :tada:

justinegeffen commented on site/content/blog/2019-12-20-onboarding-with-mattermost.md:
```diff
@@ -0,0 +1,34 @@
+I recently joined Mattermost as a Remote Software Engineer and while I’ve worked in remote environments before, I was really surprised at how amazing this company is to work with. During my onboarding I discovered that they have really developed a culture built around being remote; one that allowed for me to feel as part of a larger team and always engaged, rather than separated from my peers and ...
+
+### [Sponsored Simulation](https://docs.mattermost.com/process/developer.html#audition)
+The onboarding process at Mattermost was as unique as the company itself. As the company is fully remote, they took a great deal of care in making sure that not only was I a good fit, but that I would enjoy the environment and process that they instill. The first step was through a sponsored simulation, where I was given the opportunity to go beyond the normal open-source contributions and work d ...
```
```suggestion
The onboarding process at Mattermost was as unique as the company itself. As the company is fully remote, they took a great deal of care in making sure that not only was I a good fit, but that I would enjoy the environment and processes that they instill. The first step was through a sponsored simulation, where I was given the opportunity to go beyond the normal open-source contributions and work directly with a mentor. 
```

justinegeffen commented on site/content/blog/2019-12-20-onboarding-with-mattermost.md:
```diff
@@ -0,0 +1,34 @@
+One of the first things that occurred was that my mentor and a member of Human Resources created an extensive checklist for me to follow. This took me through all the starting documentation I would need to know and introduced me to videos and our development processes. This took me through my first two weeks and made it easy to keep on track as I learned more about my duties and the company as a  ...
+
+### [Extensive Resources](https://docs.mattermost.com/)
+There is a vast wealth of documentation available through the [Mattermost Documentation](https://docs.mattermost.com/), [Developer Documentation](https://developers.mattermost.com/), and their [Handbook](https://handbook.mattermost.com/). Everything from setting up your development environment and the Tick-Tock branching process to the setup and customization of the servers, apps, and plugins. Ma ...
```
```suggestion
There is a vast wealth of documentation available through the [Mattermost Documentation](https://docs.mattermost.com/), [Developer Documentation](https://developers.mattermost.com/), and the [Handbook](https://handbook.mattermost.com/). Everything from setting up your development environment and the Tick-Tock branching process to the setup and customization of the servers, apps, and plugins. Many of these are part of the onboarding checklist, but it will take weeks to ingest all the knowledge!
```

imisshtml commented:
Made the final two corrections, merging

hanzei commented:
@imisshtml GitHub actions failed for this PR. I've fixed the issue locally and manually re-run. The announcement is not working: https://community-release.mattermost.com/core/pl/8f9umzhpbbrk5x4ko16hfaxs3o

I've filled #518 to track the underlying issue.
